src/wp-includes/html-api/class-wp-html-tag-processor.php

```diff
@@ -711,6 +711,41 @@ public function release_bookmark( $name ) {
 	}
 
 
+	/**
+	 * Allow internal WordPress code to extract raw HTML snippets associated with a matched tag.
+	 *
+	 * This is meant for internal use to support existing filters that pass segments of input HTML
+	 * for further analysis. Filters which rely on these values should be updated to interact with
+	 * the structural methods this class provides.
+	 *
+	 * @access private
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string $internal_use_only Required to accept liability for exposing system internals through this function.
+	 * @param string $content           What portion of the raw HTML markup to return: 'entire-tag' or 'only-attributes'.
+	 * @return string|null Raw HTML markup from input document for matched tag, if matched, else NULL.
+	 */
+	public function _wp_internal_extract_raw_token_markup( $internal_use_only, $contents = 'entire-tag' ) {
+		$disclaimer = <<<INTERNAL_ONLY
+			I understand that this is only for internal WordPress usage and something
+			will likely break if used elsewhere. This function comes with no warranty.
+INTERNAL_ONLY;
+
+		if ( $internal_use_only !== $disclaimer || null === $this->tag_name_starts_at ) {
```

Comment:
This is an interesting approach. I haven't seen how we do things like this in WP - is this the first function of such nature?

Another potential idea is to prefix such functions with `unsafe` or something, like React does it :)

Reply:
Gutenberg uses this in some places. we considered `unsafe_` prefixes but I don't like those because of how normative they can become and how they can unintentionally lesson the impact of seeing `unsafe_` in the code.

this is a method I think I would prefer we never merge, and instead of supporting these old filters we move to new ones entirely. if we _do_ put these in I want to discourage people as much as possible to call them separately, but we need to make the methods public for Core's internal consumption.